stempy/blockiterator.cpp

```diff
@@ -0,0 +1,66 @@
+
+#include "blockiterator.h"
+
+#include <iostream>
+
+using std::cout;
+using std::cerr;
+using std::endl;
+
+using std::string;
+using std::vector;
+
+namespace stempy {
+
+BlockIterator::BlockIterator(const vector<string>& fileList)
```

Comment:
I was morning thinking that we would be able to iterator over the reader, something like:
```c++
StreamReader::iterator iter = reader.begin()

```

We can overload the StreamReader constructor to take a list of files.